src/main/java/views/add_edit_category_views/EditCategoryV.java

```diff
@@ -0,0 +1,111 @@
+ * View class for the EditCategoryV that extends Component class and implements ActionListener interface.
+ */
+public class EditCategoryV extends Component implements ActionListener{
+    private final CategoryC controller;
+    private final JComboBox<String> categoryCombo;
+    private final JTextField nameInput;
+    private final JTextField budgetInput;
+    private String selectedCategory;
+    private final int monthID;
+    private final SessionStorage currSession;
+    /**
+     * Builds EditCategoryV for user entries.
+     * @param controller CategoryC reacts to user input to return a CategoryOD.
+     * @param existingCategory String of existing categories in the MonthlyStorage with monthID.
+     * @param monthID int representing the MonthlyStorage.
+     * @param currSession SessionStorage the current working session.
+     */
+    public EditCategoryV(CategoryC controller, String[] existingCategory, int monthID, SessionStorage currSession) {
+        JLabel selectCategoryLabel = new JLabel(" Select existing category:");
+        this.categoryCombo = new JComboBox<>(existingCategory); // category list
+        JLabel nameLabel = new JLabel("New Category Name:");
+        this.nameInput = new JTextField(15);
+        JLabel budgetLabel = new JLabel(" New Category Budget:");
+        this.budgetInput = new JTextField(15);
+        JButton submit = new JButton("Submit");
+        submit.setSize(30,10);
+
+        JFrame frame = new JFrame();
+        frame.setDefaultCloseOperation(JFrame.HIDE_ON_CLOSE);
+        frame.setTitle("Edit Category");
+        frame.setSize(500,300);
+
+        JPanel panel = new JPanel();
+        panel.setBorder(BorderFactory.createEmptyBorder(50, 30, 50, 30));
+        panel.setLayout(new GridLayout(0,1));
+
+        panel.add(selectCategoryLabel);
+        panel.add(categoryCombo);
+        panel.add(nameLabel, BorderLayout.WEST);
+        panel.add(nameInput, BorderLayout.CENTER);
+        panel.add(budgetLabel);
+        panel.add(budgetInput);
+
+        frame.add(panel, BorderLayout.NORTH);
+        frame.pack();
+        frame.setVisible(true);
+
+        submit.addActionListener(this);
+        categoryCombo.addActionListener(this);
+
+        this.controller = controller;
+        this.monthID = monthID;
+        this.currSession = currSession;
+    }
+
+    /**
+     * Tries an Edit Category Use Case.
+     * Pop-up window with context specific message may be shown to user.
+     */
+    private void tryUseCaseEdit(){
+        CategoryOD message = null;
+        try {
+            message = controller.categoryInMonth(nameInput.getText(), String.valueOf(budgetInput.getText()), monthID, currSession, selectedCategory);
+        } catch (EntityException e) {
+            JOptionPane.showMessageDialog(this, "This month does not exist in current session. Please go to add month page.");
+        }
+        if (message != null) {
+            JOptionPane.showMessageDialog(this, message.getMessage());
```

Comment:
Ari's comment got outdated, but I think it is still relevant. 
https://github.com/CSC207-2023Y-UofT/course-project-DoorKeepers/pull/13#discussion_r1283963844

@luluwuwu 
Although you have assigned null to message, since message is a CategoryOD type, it will never be null if the try is run successfully without exception. I think this is why Ari has suggested the change written in the link above.